src/server/system_services/bucket_server.js

```diff
@@ -479,6 +479,7 @@ async function put_bucket_encryption(req) {
 async function get_bucket_policy(req) {
     dbg.log0('get_bucket_policy:', req.rpc_params);
     const bucket = find_bucket(req, req.rpc_params.name);
+    await bucket_policy_utils.validate_s3_policy_owner(bucket.owner_account._id, req.account);
```

Comment:
In case the function `validate_s3_policy_owner` arguments are strings I think you will need to convert those when passing them (in case you change the `account` to `account_id` as well).

Reply:
I am not sure about it, I could see lots of places without string conversion and its working there

Reply:
As I understand it the _id origin is from Mongo in which it was an object, so one of the ways for that comparison is converting the ID to string and then use the `===` operator if you want (anyway, it will also match the type of strings that you defined in the JSDoc.
reference I saw:
https://stackoverflow.com/questions/11637353/comparing-mongoose-id-and-strings